consensus/consensus-types/src/block.rs

```diff
@@ -111,7 +111,8 @@ impl Block {
             None => 0,
             Some(payload) => match payload {
                 Payload::InQuorumStore(pos) => pos.proofs.len(),
-                Payload::DirectMempool(txns) => txns.len(),
+                Payload::DirectMempool(_txns) => 0,
```

Comment:
why do we return 0 here?

Reply:
Because `txns.len()` and `pos.proofs.len()` are not the same. `pos.proofs.len()` is number of batches, and there is no concept of batches in `Payload::DirectMempool`

Reply:
hmm, but the fn is about payload size and we have different payload types, so depending on what type is used, we should return best effort size? Also, for this particular fn I would argue we should return the number of txns by iterating though the proofs and summing it.

Reply:
@bchocho